src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs

```diff
@@ -42,7 +42,7 @@ public InputNamespace CreateNamespace(CodeModel codeModel, SchemaUsageProvider s
                 Models: models,
                 Enums: enums,
                 ApiVersions: GetApiVersions(codeModel),
-                Auth: new CodeModelSecurity(Schemes: codeModel.Security.Schemes.OfType<SecurityScheme>().ToList()));
+                Auth: new CodeModelSecurity(Schemes: codeModel.Security.Schemes.OfType<SecurityScheme>().Distinct().ToList()));
```

Comment:
Let's make `SecuritySchemesComparer` and use it here instead of changing `Equals` of the `SecurityScheme`

Reply:
The rest LGTM.